### ISSUE TITLE
How can I use BatchNorm in a multi-GPU model

### ISSUE THREAD

**Issue by boluoweifenda** [tensorflow/tensorflow#8390]
I want to use batch normalization in the [cifar10 example](https://github.com/tensorflow/models/blob/master/tutorials/image/cifar10/cifar10_multi_gpu_train.py) in a multi-GPU structure. All variables are stored in CPU, I build 2 Queues one for training batch and one for testing batch;  5 models( 4 for training and another for testing),

 GPU_tower codes like this
![image](https://cloud.githubusercontent.com/assets/5405385/23892678/8d4c4e10-08d6-11e7-8d67-50d3bc32aec0.png)

****Problem is**: when I add tf.contrib.batch_norm layers in my NN model . How can I  reuse those batchnorm variables ? I set reuse = True and pass namescope but get valueError. Is there a simple way to make it work? I don't want to modify a lot.**

I know there are some high level framework such as [slim with the inception](https://github.com/tensorflow/models/blob/master/inception/inception/inception_train.py#L253) model with a built-in batch-norm 
but I can hardly do personal implementations (I need to add many instructions in slim.conv2d and tf.contrib.conv2d, so it's more convenient to use tf.nn.conv2d), 

**Could I just follow the inception Sync method but without high level frame like slim ?**
Thanks a lot!

**Comment by concretevitamin** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by John1231983** [comment]
Great @boluoweifenda: I have same problem as you. Did you solve it?

**Comment by Saicat** [comment]
Do not always ask us to ask on StackOverflow.
